rewrite-java/src/test/java/org/openrewrite/java/AddScopeToInjectedClassTest.java

```diff
@@ -0,0 +1,54 @@
+package org.openrewrite.java;
+
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RewriteTest;
+import org.openrewrite.test.RecipeSpec;
+import org.junit.jupiter.api.Test;
+
+import static org.openrewrite.java.Assertions.java;
+
+class AddScopeToInjectedClassTest implements RewriteTest {
+
+  /**  @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new AddScopeToInjectedClass())
+                .cycles(1);
+             //   .expectedCyclesThatMakeChanges(1);
+    }*/
+
+    @DocumentExample
+    @Test
+    void scopeRequired() {
+        rewriteRun(spec -> spec.recipe(new AddScopeToInjectedClass())
```

Comment:
You might need to add an explicit JavaParser builder here, with a classpath entry containing the annotations you use in your code samples below (depending perhaps on the version of Java used). So something like `spec.parser(JavaParser.fromJavaVersion().classpath("jar-containing-javax-inject"))`.

Right now I suspect the annotations in your code samples below are not found and added to the lossless semantic tree, which indeed if I step through with the debugger appear to be missing on the `Foo.service` field.

Instead of adding a classpath entry you can also add those annotations directly as code blocks with something like:
```java
java("""
  package javax.enterprise.context;
  @Target({ElementType.Type})
  @Retention(RetentionPolicy.RUNTIME)
  public @interface Dependent {
  }
  """),
```

Once the code snippets you pass into the unit tests are valid, as in not referring to any missing types, then you should be able to step through your recipe and see what next to change.